library/core/class.oauth2.php

```diff
@@ -349,12 +352,14 @@ protected function getSettingsFormFields() {
      * Redirect to the OAuth redirect page with a verification nonce.
      *
      * @param EntryController $sender The controller initiating the request.
-     * @param  string $state The state to pass along the OAuth2 flow.
+     * @param string $state The state to pass along the OAuth2 flow.
+     * @param string $context The context of the redirect either register or sign in.
      */
-    public function entryRedirectEndpoint(\EntryController $sender, $state = '') {
+    public function entryRedirectEndpoint(\EntryController $sender, $state = '', $context = null) {
```

Comment:
Is it safe to slap a type on this new `$context` parameter?

Reply:
I would rather you just put the context inside the state.